internal/provider/resource_namespace_file.go

```diff
@@ -0,0 +1,200 @@
+package provider
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"strings"
+)
+
+func resourceNamespaceFile() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a Kestra Namespace File.",
+
+		CreateContext: resourceNamespaceFileCreate,
+		ReadContext:   resourceNamespaceFileRead,
+		DeleteContext: resourceNamespaceFileDelete,
+		Schema: map[string]*schema.Schema{
+			"tenant_id": {
+				Description: "The tenant id.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"namespace": {
+				Description: "The Namespace file namespace.",
```

Comment:
```suggestion
				Description: "The namespace of the namespace file.",
```
seems a little clearer